### PR TITLE
Cleanup ObjectManager usage - Magento_Developer

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This PR cleanups ObjectManager usage in Magento_Developer, adds additional dependencies with backward compatibility to api classes, and removes object manager instantiation from non-api classes.

### Fixed Issues (if relevant)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php [hunk 1]
```diff
@@ -42,32 +42,34 @@ class PreprocessorStrategy implements PreProcessorInterface
...
      * @param FrontendCompilation $frontendCompilation
      * @param ScopeConfigInterface $scopeConfig
+     * @param State|null $state
      */
     public function __construct(
         AlternativeSourceInterface $alternativeSource,
         FrontendCompilation $frontendCompilation,
-        ScopeConfigInterface $scopeConfig
+        ScopeConfigInterface $scopeConfig,
+        ?State $state = null
```
Comment (ihor-sviziev) [inline app/code/Magento/Developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php]: please make this parameter not optional as this class isn't marked with api tag

Comment (ihor-sviziev) [inline app/code/Magento/Developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php:54]: please make this argument not optional as this class isn't marked with api tag

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php (hunk 1 above)
Comment (Bartlomiejsz) [reply app/code/Magento/Developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php:54]: Hmm, isn't it? https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php#L19

Comment (ihor-sviziev) [reply app/code/Magento/Developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php:54]: Ah. sorry, I missed it

**Review by ihor-sviziev — DISMISSED** [review]
Failing unit & functional tests seems not related to changes from this PR

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7238](https://jira.corp.magento.com/browse/ENGCOM-7238) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-7238](https://jira.corp.magento.com/browse/ENGCOM-7238) has been created to process this Pull Request

**Pull request merged**